### PR TITLE
[#21587] Unsupported light theme on Keycard profile key pair migration

### LINKED ISSUE #21586: Keycard migration crashes the app on Android
# Bug Report

## Problem 
[comment]: # (A problem description)

After setting the pin for the keycard and having to scan it again, the app crashes. See the recording below.
The logcat can be found here:

[logcat_output.txt](https://github.com/user-attachments/files/17646741/logcat_output.txt)

https://github.com/user-attachments/assets/575c4f10-9965-4b51-80ee-8afc0d9e1b9e

#### Expected behavior
no crash when migrating profile to keycard

#### Actual behavior
crash 

### Reproduction

1. Create profile
2. Backup the seedphrase
3. Go to settings and have an empty/new keycard
4. Enable the feature flag
5. Go to "Keycard"
6. Press "Setup Keycard"
7. Scan the keycard
8. Enter seedphrase
9. Enter password
10. Enter PIN
11. CRASH

## Additional Information- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #21587: Unsupported light theme on Keycard profile key pair migration
# Bug Report
https://github.com/status-im/status-mobile/pull/21467

## Problem 
[comment]: # (A problem description)

https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard-UX?node-id=119-16075&node-type=frame&m=dev

We have designs only for Dark theme then the Light one is not meant to exist

### Reproduction

1. Login to app
2. Turn on Keycard feature flag
3. Go to Keycard Profile menu

#### Actual behavior

Image

#### Expected behavior

Image

https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard-UX?node-id=119-16075&node-type=frame&m=dev
### Additional Information

- Status version: PR build https://github.com/status-im/status-mobile/pull/21467
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #21601: "Keycard is not empty" screen appears at the wrong place of migration flow
# Bug Report
https://github.com/status-im/status-mobile/pull/21467

## Problem 
[comment]: # (A problem description)
User has an empty Keycard and NOT backuped Profile.
In the middle of the migration flow user closes `Ready to migrate profile` screen and then starts flow one more time then his keycard is no longer empty but no check on this till we come to the PIN entering step. But check if our Keycard ready or not should be right after `Check your card screen` ( at the beginnig of the flow) and not at the end when we enter PIN.
Please, look at the video to observe my test scenario.

#### Actual behavior

* will put timestamps as I had some long issues with card scanning

00:27 - I cut my migration flow and return to the beginning
01:10 - Here should be a check that my Keycard is not empty
01:37 - `Keycard is not empty` screen 

https://github.com/user-attachments/assets/0855ed7b-4843-4287-a02e-d0efd6771706

#### Expected behavior

Check that my `Keycard is not empty`screen should be after `Check your card` screen

Image

https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard-UX?node-id=3-89231&node-type=canvas&t=jRHi23c8KvoA2g3u-0

### Reproduction

0. Test data: empty Keycard and not backup-ed Profile
1. Login to App
2. Turn on Keycard feature flag
3. Make sure your Keycard is empty and your Profile is NOT backup-ed
4. Go through the migration flow and close the screen when you'll see `Ready to migrate your Profile` screen
5. Go through the flow one more time

### Additional Information

- Status version: (https://github.com/status-im/status-mobile/issues/21318)
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #21587
fixes #21586
fixes #21601

### CODE REVIEW

File: src/status_im/contexts/keycard/migrate/events.cljs [hunk 1]
```diff
@@ -75,15 +75,15 @@
...
              {:on-success #(rf/dispatch [:keycard/migration.phrase-entered %])}]]
            [:dispatch
-            [:open-modal :screen/backup-recovery-phrase
+            [:open-modal :screen/backup-recovery-phrase-dark
              {:on-success #(rf/dispatch [:keycard/migration.phrase-backed-up %])}]])]}))
 
 (rf/reg-event-fx :keycard/migration.phrase-entered
  (fn [{:keys [db]} [{:keys [phrase]}]]
-   {:db (assoc-in db [:keycard :migration :masked-phrase] (security/mask-data phrase))
+   {:db (assoc-in db [:keycard :migration :masked-phrase] phrase)
```
Comment (clauxx) [inline src/status_im/contexts/keycard/migrate/events.cljs:88]: :+1: 

should include this issue in the PR description as well https://github.com/status-im/status-mobile/issues/21586

**Review by clauxx — APPROVED** [review]
looks good

**Review by Parveshdhull — APPROVED** [review]

File: src/status_im/contexts/keycard/migrate/events.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/contexts/keycard/migrate/events.cljs:88]: thanks, typo

**Comment by VolodLytvynenko** [comment]
currently, we are blocked from testing it until ios builds are fixed

**Comment by mariia-skrypnyk** [comment]
Hi @flexsurfer !

Regarding https://github.com/status-im/status-mobile/issues/21587.
We have one place where some leftovers take place
iOS vs Android

Screenshot 2024-11-12 at 13 35 25

When migration is done scanning bottom sheet looks correct:

Screenshot 2024-11-12 at 13 35 45

https://github.com/status-im/status-mobile/issues/21586 ✅
https://github.com/status-im/status-mobile/issues/21601 ✅

**Comment by flexsurfer** [comment]
hey @mariia-skrypnyk thank you , could you please create a new separate task for connection sheet.

**Comment by mariia-skrypnyk** [comment]
Hey @flexsurfer !

Thanks for your great work.
Failed e2e are not related.
PR can be merged!

**Pull request merged**